ios/lockdown_assistivetouch.go

```diff
@@ -0,0 +1,52 @@
+package ios
+
+import log "github.com/sirupsen/logrus"
+import "fmt"
+
+const accessibilityDomain = "com.apple.Accessibility"
+const assistiveTouchKey  = "AssistiveTouchEnabledByiTunes"
+
+// EnableAssistiveTouch creates a new lockdown session for the device and enables or disables
+// AssistiveTouch (the on-screen software home button), by using the special key AssistiveTouchEnabledByiTunes.
+// Setting to true will enable AssistiveTouch
+// Setting to false will disable AssistiveTouch, regardless of whether it was previously-enabled through
+// a non-iTunes-related method.
+func SetAssistiveTouch(device DeviceEntry, enabled bool) error {
+	lockDownConn, err := ConnectLockdownWithSession(device)
+	if err != nil {
+		return err
+	}
+	log.Debugf("Setting %s: %t", assistiveTouchKey, enabled)
+	defer lockDownConn.Close()
+	err = lockDownConn.SetValueForDomain(assistiveTouchKey, accessibilityDomain, enabled)
+	return err
+}
+func GetAssistiveTouch(device DeviceEntry) (bool, error) {
+	lockDownConn, err := ConnectLockdownWithSession(device)
+
+	if err != nil {
+		return false, err
+	}
+	defer lockDownConn.Close()
+
+	enabledIntf, err := lockDownConn.GetValueForDomain(assistiveTouchKey, accessibilityDomain)
+
+	if err != nil {
+		return false, err
+	}else if enabledIntf == nil{
+		// In testing, nil was returned in only one case, on an iOS 14.7 device that should already have been paired.
+		// Calling SetAssistiveTouch() directly returned the somewhat more useful error: SetProhibited
+		// After re-running "go-ios pair", full functionality returned.
+		return false, fmt.Errorf("Received null response when querying %s.%s. Try re-pairing the device.", accessibilityDomain, assistiveTouchKey)
+	}
+	enabledUint64, ok := enabledIntf.(uint64)
+	if !ok {
+		// On iOS 10.x at least, "false" is returned, perhaps for any key at all.  Attempting to manipulate AssistiveTouchEnabledByiTunes had no effect
+		return false, fmt.Errorf("Expected unit64 0 or 1 when querying %s.%s, but received %T:%+v. Is this device running iOS 11+.", accessibilityDomain, assistiveTouchKey, enabledIntf, enabledIntf)
```

Comment:
I think it would be cool if you could add a check in the beginning for the iOS version. 
If we already know it doesn't work on iOS10 maybe better to fail fast?
In either case, including the device version in the error message could be good. 

Here is an example how I work with ios versions: https://github.com/danielpaulus/go-ios/blob/0e140a0f03560ce1302f00e70348796cd525dfd0/ios/imagemounter/imagedownloader.go#L56

wdyt?

Reply:
I've gone ahead and added a fail-fast check, with a more to-the-point error message, on the command-line invocation of go-ios.    However, I also added a "--force" override to go ahead and try the operations anyway, since I've only tested on a single device older than iOS 11 (and have not tested on iPad).  At least for a little while, it would be nice to *try* this method on older hardware.

I left the "Is this device running iOS 11+?" in the library call error message, with the idea that this provides some feedback for non-command-line invocations without making it impossible to invoke the function if you want to try.  
Presumably if you are using this in a library with a mechanism to automate assistive touch, you should have written in custom case-checking for older devices yourself anyway (presuming you intend to support iOS 10 and older...). 

What do you think?

Reply:
Perfect.